### PR TITLE
Support Python 3.12

### PR DESCRIPTION
**Pull request by mbargull**
Fixes gh-689

This pulls in the code from https://github.com/anaconda-graveyard/clyent/blob/02b1eeb23a02c7d1ec64262c187eac013954fa45/clyent/__init__.py#L32-L143 with only minimal adjustments.

https://github.com/Anaconda-Platform/anaconda-client/commit/c3150186451dda38284186700e8db9b0cc9aa520 is actually a verbatim copy of that code.

N.B.: If code style, annotation, or other changes are desired, please feel free to push to this PR's branch (i.e., "Allow edits by maintainers" is checked).

### CODE REVIEW

File: binstar_client/scripts/cli.py [hunk 1]
```diff
@@ -49,6 +56,121 @@ def file_or_token(value: str) -> str:
...
+    return grp_data
+
+
+class json_help(argparse.Action):
+    def __init__(self, nargs=0, help=argparse.SUPPRESS, **kwargs):
+        argparse.Action.__init__(self, nargs=nargs, help=help, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string):
+        self.nargs = 0
+        docs = {'prog': parser.prog,
```
Comment (jezdez) [inline binstar_client/scripts/cli.py:101]: I'm not sure what the code formatting policy is for anaconda-client, but that looks weird.

**Review by jezdez — APPROVED** [review]

File: binstar_client/scripts/cli.py (hunk 1 above)
Comment (mbargull) [reply binstar_client/scripts/cli.py:101]: Yep, no idea either.
> \[...\] verbatim copy of that code.
> 
> N.B.: If code style, annotation, or other changes are desired, please feel free to push to this PR's branch \[...\]

Comment (mbargull) [reply binstar_client/scripts/cli.py:101]: (And thanks for reviewing!)

Comment (mbargull) [reply binstar_client/scripts/cli.py]: CI log indicated which linters are used.
Commits to make those happy have been added.

**Pull request merged**